src/components/intersection.js

```diff
@@ -57,75 +59,112 @@ AFRAME.registerComponent('intersection', {
     function createSidewalkElem({
       length,
       width,
+      radius,
       positionVec,
       scaleVec = { x: 1, y: 1, z: 1 },
       rotationVec,
       displayName
     }) {
       const sd = document.createElement('a-entity');
-      // every 2 meters repeat sidewalk texture
-      const repeatCountInter = [width / 2, parseInt(length / 2)];
+      // Radius should not be greater than any side
+      const boundedRadius = Math.min(radius, length, width);
+
+      const points = [];
+      points.push(new THREE.Vector2(0, 0));
+      points.push(new THREE.Vector2(length, 0));
+      // Only if a radius is set, create the arc
+      if (radius > 0) {
+        const arc = new THREE.EllipseCurve(
+          length - boundedRadius,
+          width - boundedRadius,
+          boundedRadius,
+          boundedRadius,
+          0,
+          Math.PI / 2
+        );
+        points.push(...arc.getSpacedPoints());
+      } else {
+        points.push(new THREE.Vector2(length, width));
+      }
+      points.push(new THREE.Vector2(0, width));
 
-      sd.setAttribute(
-        'geometry',
-        `primitive:box; depth: ${length}; width: ${width}; height: 0.4`
+      // Create new shape out of the points:
+      const curbShape = new THREE.Shape(points);
+      const curbGeom = new THREE.ExtrudeGeometry(curbShape, {
+        depth: 0.4, // Match existing sidewalk thickness
+        bevelEnabled: false
+      });
+
+      const sidewalkTexture = new THREE.TextureLoader().load(
+        document.getElementById('seamless-sidewalk').src
       );
+      console.log(sidewalkTexture);
+      sidewalkTexture.wrapS = THREE.RepeatWrapping;
+      sidewalkTexture.wrapT = THREE.RepeatWrapping;
+      sidewalkTexture.repeat.set(0.5, 0.5); // Scale the texture to repeat twice every meter
+
+      const material = new THREE.MeshStandardMaterial({
+        map: sidewalkTexture,
+        roughness: 0.8, // Same roughness as sidewalk mixin in src/assets.js
+        color: 0xcccccc // Darkens the texture to match existing sidewalk
+      });
+      const mesh = new THREE.Mesh(curbGeom, material);
+      mesh.scale.setX(scaleVec.x);
+      mesh.scale.setY(scaleVec.y);
+      mesh.scale.setZ(scaleVec.z);
+
```

Comment:
Because you create the three geometry and material yourself, the dispose of them are not handled by aframe, you need to take care of it yourself in `remove()`, assign the variables on `this` so you can then add:

```js
remove() {
  if (this.curbGeom) this.curbGeom.dispose();
  if (this.material) this.material.dispose();
}
```

The material previously could be shared among several meshes so for several intersections, that's a feature of aframe it does that automatically. Here we create one for each intersection. We could I guess use a global variable in this module to create only once the material if not currently exist (and not dispose of it in `remove()`).

Reply:
Was not aware of that, thanks for the tip! I will add the global in `intersection.js` to avoid creating it multiple times then.

Reply:
Yes, your changes are perfect. When I commented, I didn't pay attention at first we had 4 sidewalks in the intersection, so yes sharing the material for the 4 created sidewalks was a must indeed.
I was talking about having several intersections in the scene, we could have the material shared for all the sidewalks of all the intersections, but the hard part is to dispose the material only when all the intersections are removed. You can usually do that with an aframe system that do this reference counting, that's what the aframe material system is actually doing to dispose automatically the material when no other aframe entities is using it. But yeah maybe too much for this PR, and I don't think we'll have a lot of intersections in a single scene, probably only one or two. We can revisit later if we need some performance.

Reply:
That makes sense. Thanks for the tip!